test-suite/buildtest.sh

```diff
@@ -16,6 +16,16 @@
 action="${1}"
 config="${2}"
 
+# Allow a layer to enable optional features supported in current build environment
+if ${PKG_CONFIG:-pkg-config} --exists --exact-version=1.0.0 libecap 2>/dev/null
+then
+    CONFIGURE_FLAGS_MAYBE_ENABLE_ECAP="--enable-ecap"
+fi
+if ${PKG_CONFIG:-pkg-config} --exists  valgrind 2>/dev/null
+then
+    CONFIGURE_FLAGS_MAYBE_ENABLE_VALGRIND="--with-valgrind-debug"
+fi
+
```

Comment:
I think we need only these two optional features out of the list of optional features that are commented out in layer-* files. Others (such as gnutls) are enabled by default:

```
configure: Library 'gnutls' support: yes (auto)  -lgnutls
```
and may be disabled by the corresponding configure option (e.g., --without-gnutls).

Reply:
Understood. I have suggested the corresponding documentation adjustment, to emphasize that we want to handle default-disabled features here.

It would be good to test that ./configure actually does enable those default-enabled features when they should be enabled, but that enhancement can be viewed as lying outside this PR scope.